### PR TITLE
use android test because roboelectric tests are failing

### PR DESCRIPTION
**Pull request by epicadk**

### CODE REVIEW

**Comment by jingtang10** [comment]
@ktarasenko - can you take a look at this please? thanks

**Comment by jingtang10** [comment]
should we be replacing all robolectric tests runners with androidx test runners?

**Comment by epicadk** [comment]
> should we be replacing all robolectric tests runners with androidx test runners?

The error was because the connection to the database is not closed after the test is run. Not sure how we can achieve this. Also do you think it makes sense to add an `useInMemory` property to fhirConfig for testing purposes?

**Comment by jingtang10** [comment]
> > should we be replacing all robolectric tests runners with androidx test runners?
> 
> The error was because the connection to the database is not closed after the test is run. Not sure how we can achieve this. Also do you think it makes sense to add an `useInMemory` property to fhirConfig for testing purposes?

the title of this pr is use android test instead of robolectric. is this inaccurate then?

**Comment by epicadk** [comment]
> > > should we be replacing all robolectric tests runners with androidx test runners?
> > 
> > 
> > The error was because the connection to the database is not closed after the test is run. Not sure how we can achieve this. Also do you think it makes sense to add an `useInMemory` property to fhirConfig for testing purposes?
> 
> the title of this pr is use android test instead of robolectric. is this inaccurate then?

It is accurate but only the the ak/fhir_dal branch because I don't have write access anymore. Otherwise this would've just been another commit on #1273

**Review by ktarasenko — APPROVED** [review]
LGTM 

I had to find a workaround for database errors in tests in #1280.  I'd rather not to make FhirEngineProvider.kt a singleton. WDYT?

**Comment by epicadk** [comment]
> LGTM
> 
> I had to find a workaround for database errors in tests in #1280. I'd rather not to make FhirEngineProvider.kt a singleton. WDYT?

Would using Android Test Orchestrator be a possible fix for this?

File: workflow/src/main/java/com/google/android/fhir/workflow/FhirEngineDal.kt [hunk 1]
```diff
@@ -51,7 +51,7 @@ class FhirEngineDal(private val fhirEngine: FhirEngine) : FhirDal {
     }
   }
 
-  override fun search(resourceType: String): MutableIterable<IBaseResource> {
+  override fun search(resourceType: String): Iterable<IBaseResource> {
```
[the remark this answers is not in the document]
Comment (epicadk) [reply workflow/src/main/java/com/google/android/fhir/workflow/FhirEngineDal.kt:54]: Yeah I think I'll revert them.

**Comment by ktarasenko** [comment]
@epikadk, not really, i added a way to reset FhirEngineProvider. I don't think it has to be singleton at all

**Comment by jingtang10** [comment]
approved and merged since it's PR into your development branch

**Pull request merged**